tests/test_s3_to_json_s3.py

```diff
@@ -0,0 +1,477 @@
+            assert file_name in actual_file_names
+
+    def test_get_dataset_identifier(self, schema_mapping_uri, dataset_mapping_uri):
+        schema_mapping = get_data_mapping(data_mapping_uri=schema_mapping_uri)
+        data_mapping = get_data_mapping(data_mapping_uri=dataset_mapping_uri)
+        json_schema = {
+                "$id": ("https://sage-bionetworks.github.io/mobile-client-json/"
+                        "schemas/v2/ArchiveMetadata.json")
+        }
+        schema_dataset_identifier = get_dataset_identifier(
+                json_schema=json_schema,
+                schema_mapping=schema_mapping,
+                dataset_mapping=data_mapping,
+                file_metadata={}
+        )
+        assert schema_dataset_identifier == "ArchiveMetadata_v1"
+        file_metadata = {
+                "assessment_id": "dccs",
+                "assessment_revision": "5",
+                "file_name": "motion.json",
+                "app_id": "mobile-toolbox"
+        }
+        data_dataset_identifier = get_dataset_identifier(
+                json_schema={ "schema": None },
+                schema_mapping=schema_mapping,
+                dataset_mapping=data_mapping,
+                file_metadata=file_metadata
+        )
+        assert data_dataset_identifier == "MotionRecord_v1"
+
+    def test_validate_against_schema(self, s3_obj, metadata_json_schema):
+        file_metadata = {
+                "assessment_id": s3_obj["Metadata"]["assessmentid"],
+                "assessment_revision": s3_obj["Metadata"]["assessmentrevision"],
+                "file_name": "metadata.json",
+                "app_id": "mobile-toolbox"
+        }
+        with zipfile.ZipFile(io.BytesIO(s3_obj["Body"])) as z:
+            with z.open(file_metadata["file_name"], "r") as p:
+                j = json.load(p)
+                all_errors = validate_against_schema(
+                        data=j,
+                        schema=metadata_json_schema,
+                        base_uri="https://sage-bionetworks.github.io/mobile-client-json/schemas/v2"
+                )
+                assert isinstance(all_errors, list)
+                incorrect_metadata_json_schema = metadata_json_schema
+                incorrect_metadata_json_schema["properties"]["cookies"] = { "type": "string" }
+                incorrect_metadata_json_schema["required"] = \
+                        incorrect_metadata_json_schema["required"] + ["cookies"]
+                all_errors = validate_against_schema(
+                        data=j,
+                        schema=incorrect_metadata_json_schema,
+                        base_uri="https://sage-bionetworks.github.io/mobile-client-json/schemas/v2"
+                )
+                assert len(all_errors) == 1
+
+    def test_validate_data(self, s3_obj, archive_map, dataset_mapping_uri):
+        dataset_mapping = get_data_mapping(data_mapping_uri=dataset_mapping_uri)
+        validation_result = validate_data(
```

Comment:
NIt: Think about how once you've tested functions already, then you can mock the other functions.  That said, mocking requires maintenance.

Reply:
I'm using local fixtures to pass these mappings into the test functions now.